src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -94,6 +95,19 @@ public static function fromDsn(string $dsn, array $options = [], HttpClientInter
         if (isset($parsedUrl['query'])) {
             parse_str($parsedUrl['query'], $query);
         }
+
+        // check for extra keys in options
+        $optionsExtraKeys = array_diff(array_keys($options), array_keys(self::DEFAULT_OPTIONS));
+        if (0 < \count($optionsExtraKeys)) {
+            throw new InvalidArgumentException(sprintf('Unknown option found: [%s]. Allowed options are [%s].', implode(', ', $optionsExtraKeys), implode(', ', array_keys(self::DEFAULT_OPTIONS))));
+        }
+
+        // check for extra keys in options
+        $queryExtraKeys = array_diff(array_keys($query), array_keys(self::DEFAULT_OPTIONS));
```

Comment:
wouldn't this move cause a bug due to moving this before you unset `$query['sslmode']` ?

Reply:
sorry for the mess in that PR, I pushed-force several version.

I removed the unset, and added the sslmode in the list of allowed parameters 

(in all case, I added tests that assert that sslmode option won't trigger an exception anymore)